### ISSUE TITLE
I cannot send mail from my mail server

### ISSUE THREAD

**Issue by keystroke3** [modoboa/modoboa#2030]
# Impacted versions
* Modoboa: 1.16.1
* installer used: Yes
* Webserver: Nginx

# Steps to reproduce
Run installer
Added `mydomain.tld` in the admin panel
Added super admin user `admin@mydomain.tld`
Deleted default admin user
Added SPF and DMARC records to domain registrar using Modoboa generated DKIM form `mail.mydomain.tld`
Added a simple user `user` in identities.
Restart the VPS
Run `sudo dpkg -P clamav-daemon clamav-base clamav-freshclam` to remove ClamAV
Restart VPS again
Logged out of `admin@mydomain.tld` to `user@mydomain.tld`
wrote test emails to an existing gmail address
wrote test emails from said gmail address to `user@mydomain.tld`
wrote an email from `user@mydomain.tld` to a non existing address
try to add `user@mydomain.tld` to Thunderbird
# Current behavior
Email sent from gmail address are received by `user@mydomain.tld`
Email sent from `user@mydomain.tld` to non existing address got a message sending failed response.
Email sent form `user@mydomain.tld` to existing gmail address not received at all.
Signing in to Thunderbird detects the mailbox settings but hangs on `checking password` and eventually fails.
Log file at  `/var/log/nginx/mail.mydomain.tld-error.log` is blank.
Spam folder for destination gmail address is also empty
# Expected behavior
Emails sent from and to `user@mydomain.tld` to arrive  and to be able to sign in into Thunderbird with it.

**Comment by MrGeneration** [comment]
The errors you're looking for should be living within either `/var/log/mail.log` or `/var/log/mail.err`.
Your issues are mail server not web server related. :)

It's possible that your automx provides invalid configuration data to your thunderbird.
So you may double check the detected settings.

In my case the following does work:
- IMAP Port 993 with SSL/TLS & password, normal
- SMTP Port 587 with STARTTLS & password, normal

**Comment by keystroke3** [comment]
Thanks @MrGeneration  for the reply.
> The errors you're looking for should be living within either /var/log/mail.log or /var/log/mail.err.

Thanks for the reply. I looked into the logs and found that there is a potion of the logs that keep repeating:
```
Oct 10 23:31:49 salvaje postfix/qmgr[1012237]: 44364279D98: from=<mail@mydomain1.com>, size=1191, nrcpt=1 (queue active)
Oct 10 23:31:49 salvaje postfix/qmgr[1012237]: 323B5279A62: from=<ted@mydomian2.com>, size=1014, nrcpt=1 (queue active)
Oct 10 23:32:19 salvaje postfix/smtp[1191190]: connect to gmail-smtp-in.l.google.com[64.233.184.26]:25: Connection timed out
Oct 10 23:32:19 salvaje postfix/smtp[1191191]: connect to gmail-smtp-in.l.google.com[64.233.184.26]:25: Connection timed out
Oct 10 23:32:49 salvaje postfix/smtp[1191190]: connect to alt1.gmail-smtp-in.l.google.com[209.85.233.26]:25: Connection timed out
Oct 10 23:32:49 salvaje postfix/smtp[1191191]: connect to alt1.gmail-smtp-in.l.google.com[209.85.233.26]:25: Connection timed out
Oct 10 23:33:19 salvaje postfix/smtp[1191190]: connect to alt2.gmail-smtp-in.l.google.com[172.253.118.26]:25: Connection timed out
Oct 10 23:33:19 salvaje postfix/smtp[1191191]: connect to alt2.gmail-smtp-in.l.google.com[172.253.118.26]:25: Connection timed out
Oct 10 23:33:49 salvaje postfix/smtp[1191190]: connect to alt3.gmail-smtp-in.l.google.com[108.177.97.27]:25: Connection timed out
Oct 10 23:33:49 salvaje postfix/smtp[1191191]: connect to alt3.gmail-smtp-in.l.google.com[108.177.97.27]:25: Connection timed out
Oct 10 23:34:19 salvaje postfix/smtp[1191190]: connect to alt4.gmail-smtp-in.l.google.com[74.125.28.26]:25: Connection timed out
Oct 10 23:34:19 salvaje postfix/smtp[1191191]: connect to alt4.gmail-smtp-in.l.google.com[74.125.28.26]:25: Connection timed out
Oct 10 23:34:19 salvaje postfix/smtp[1191190]: 44364279D98: to=<[testemail]@gmail.com>, relay=none, delay=292717, delays=292566/0.07/150/0, dsn=4.4.1, status=deferred (connect to alt4.gmail-smtp-in.l.google.com[74.125.28.26]:25: Connection timed out)
```
I am guessing that the repeating section is for every time I tried sending an email from my server. What could be causing the time out issues? I tried turning off my `ufw` but there was no difference.

>It's possible that your automx provides invalid configuration data to your thunderbird

Could you please point me to a resource that could help me sort this issue out?

**Comment by MrGeneration** [comment]
Please ensure that port 25 is open for outgoing and incoming connections.
Also, make sure Port 25 is *not* blocked by your provider / hoster.

There are providers and hosters that do restrict these ports by default. Outgoing connection timeouts are an indicator for that.

**Comment by keystroke3** [comment]
Port 25 seems to be open. When I run `telnet mydomian.com 25` on my local machine, this is the output:
```
Trying <my server ip> ...
Connected to mydomain.com.
Escape character is '^]'.
220-Welcome, please wait...
220 mail.mydomain.com ESMTP
```  
When I run `netstat -tunlp` to check for listening ports on the server, this is what I get:
```
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 127.0.0.1:9998          0.0.0.0:*               LISTEN      1665/amavisd-new (m
tcp        0      0 127.0.0.1:9999          0.0.0.0:*               LISTEN      1240/python
tcp        0      0 0.0.0.0:143             0.0.0.0:*               LISTEN      665/dovecot
tcp        0      0 0.0.0.0:8080            0.0.0.0:*               LISTEN      1542372/apache2
tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      1542370/nginx: mast
tcp        0      0 127.0.0.1:5232          0.0.0.0:*               LISTEN      1244/python3
tcp        0      0 0.0.0.0:8081            0.0.0.0:*               LISTEN      1542372/apache2
tcp        0      0 0.0.0.0:8082            0.0.0.0:*               LISTEN      1542372/apache2
tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN      524/systemd-resolve
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      740/sshd: /usr/sbin
tcp        0      0 127.0.0.1:5432          0.0.0.0:*               LISTEN      815/postgres
tcp        0      0 0.0.0.0:25              0.0.0.0:*               LISTEN      2607/master
tcp        0      0 127.0.0.1:12345         0.0.0.0:*               LISTEN      1269/opendkim
tcp        0      0 0.0.0.0:443             0.0.0.0:*               LISTEN      1542370/nginx: mast
tcp        0      0 0.0.0.0:4190            0.0.0.0:*               LISTEN      665/dovecot
tcp        0      0 0.0.0.0:993             0.0.0.0:*               LISTEN      665/dovecot
tcp        0      0 127.0.0.1:10024         0.0.0.0:*               LISTEN      1665/amavisd-new (m
tcp        0      0 127.0.0.1:10025         0.0.0.0:*               LISTEN      2607/master
tcp        0      0 127.0.0.1:10026         0.0.0.0:*               LISTEN      1665/amavisd-new (m
tcp        0      0 0.0.0.0:587             0.0.0.0:*               LISTEN      2607/master
tcp        0      0 127.0.0.1:6379          0.0.0.0:*               LISTEN      817/redis-server 12
tcp        0      0 127.0.0.1:11211         0.0.0.0:*               LISTEN      675/memcached
tcp6       0      0 ::1:9998                :::*                    LISTEN      1665/amavisd-new (m
tcp6       0      0 :::143                  :::*                    LISTEN      665/dovecot
tcp6       0      0 :::80                   :::*                    LISTEN      1542370/nginx: mast
tcp6       0      0 ::1:5232                :::*                    LISTEN      1244/python3
tcp6       0      0 :::22                   :::*                    LISTEN      740/sshd: /usr/sbin
tcp6       0      0 :::443                  :::*                    LISTEN      1542370/nginx: mast
tcp6       0      0 :::4190                 :::*                    LISTEN      665/dovecot
tcp6       0      0 :::993                  :::*                    LISTEN      665/dovecot
tcp6       0      0 ::1:10024               :::*                    LISTEN      1665/amavisd-new (m
tcp6       0      0 ::1:10026               :::*                    LISTEN      1665/amavisd-new (m
tcp6       0      0 ::1:6379                :::*                    LISTEN      817/redis-server 12
udp        0      0 0.0.0.0:33507           0.0.0.0:*                           1240/python
udp        0      0 0.0.0.0:49163           0.0.0.0:*                           1199/uwsgi
udp        0      0 127.0.0.53:53           0.0.0.0:*                           524/systemd-resolve
udp        0      0 0.0.0.0:53810           0.0.0.0:*                           2424/amavisd-new (c

```

Are there any other tests I can run to try and narrow down the issue?

**Comment by MrGeneration** [comment]
Port 25 is probably blocked on transit between your host and the other.
This may be dependent on your hosting provider, whoever that is.

You should check with their FAQ or support.

**Comment by keystroke3** [comment]
After more digging, I found my local IP address is blocked by zen.spamhaus.org. I read through [their help page](https://www.spamhaus.org/faq/section/Spamhaus%20PBL):

> Help! My IP address is on the PBL!
If a mail program such as Apple Mail, Thunderbird, or Outlook  is being used, and the sending of email is failing due to a
PBL listing,  the solution is to ensure that SMTP Authentication is enabled and  working correctly: this will immediately correct the problem. Please see  Enabling SMTP Authentication for more information.

How do I set this up SMTP authentication properly on Modoboa? I used to use Mailcow and did not experience this issue. Could there be a config setting that I need to change?

**Comment by MrGeneration** [comment]
Sorry but this as far as I can help.

**Comment by keystroke3** [comment]
I found that the SMTP authentication [section in the docs](https://modoboa.readthedocs.io/en/latest/configuration.html#smtp). Now I am wondering if it is possible to empty the database and start a fresh without reinstalling modoboa. I am having issues with an identity that I created but is not visible in the admin panel. SMTP is trying to authenitcate with that email, and is bringing issues.

**Comment by keystroke3** [comment]
Ok, I made some progress, I deleted the entire domain from the list and the authentication is working now with my good domain. The logs however, show that there is an ip that is blacklisted, but it's not mine. It's not the vps either, though a whois lookup shows that it is a cloud hosting company. I will contact DigitalOcean support tomorrow and see if the problem is with them.
